lib/internal/bootstrap/loaders.js

```diff
@@ -97,6 +97,78 @@ const internalBindingWhitelist = new SafeSet([
   'zlib'
 ]);
 
+{
+  const { safeToString } = getInternalBinding('util');
+  const { setPrepareStackTraceCallback } = getInternalBinding('errors');
+  let wrapper;
+  const ErrorConstructor = Error;
+  setPrepareStackTraceCallback((globalThis, error, trace, message) => {
+    // Polyfill of V8's Error.prepareStackTrace API.
+    // By owning this call, we can prevent error double-decoration.
+    if (typeof globalThis.Error.prepareStackTrace === 'function') {
+      // https://crbug.com/v8/7848
+      return globalThis.Error.prepareStackTrace(error, trace);
+    }
+    if (typeof ErrorConstructor.prepareStackTrace === 'function') {
+      return ErrorConstructor.prepareStackTrace(error, trace);
+    }
+    const errorString = safeToString(error);
+    if (trace.length === 0) {
+      // If the trace length is 0, V8 cannot capture a throw site.
+      // `sourceLine` below will be the line where `.stack` was accessed,
+      // which is rather nonsensical. Instead we will do what V8 does, and
+      // just return the raw error string.
+      return errorString;
+    }
+    const [
+      resourceName,
+      lineNumber,
+    ] = message;
+    let [
+      ,, // Split up because of let/const eslint
+      startPosition,
+      endPosition,
+      sourceLine,
+    ] = message;
+    // The default formatting that V8 uses is:
```

Comment:
I am confused...doesn't v8 already do this now? Those are tested in `test/message/fail` under v8..

Reply:
Oh, right, looks like it's done by d8, but it's done in the AddMessageListenerWithErrorLevel callback instead. I wonder why we have to do this in the prepareStackTrace callback?

Reply:
this changes it to decorate all errors instead of 
just ones that were lucky enough to get decorate called on them. that's why the asserts are failing. this wasn't the primary motivation for this change though, see the original comment for more info.